dqc_us_rules/dqc_us_0001.py

```diff
@@ -0,0 +1,343 @@
+
+    :param axis: The axis to check
+    :type axis: :class:'~arelle.ModelDTSObject.ModelConcept'
+    :param axis_config: The axis-specific config.
+    :type axis_config: dict
+    :param relset: The relationshipSet for the axis.
+    :type relset: set
+    :param val: val from which to gather end dates
+    :type val: :class:'~arelle.ModelXbrl.ModelXbrl'
+    :param role: The role for the relationship
+    :type role: str
+    :return: No direct return
+    :rtype: None
+    """
+    _run_member_checks(axis, axis_config, relset, val, role)
+    _run_extension_checks(axis, axis_config, relset, val, role)
+
+
+def _run_member_checks(axis, axis_config, relset, val, role):
+    """
+    Run the checks on included and excluded members and companion axes.
+    Extensions are not checked.  Error as appropriate.
+
+    :param axis: The axis to check
+    :type axis: :class:'~arelle.ModelDTSObject.ModelConcept'
+    :param axis_config: The axis-specific config.
+    :type axis_config: dict
+    :param relset: The relationshipSet for the axis.
+    :type relset: set
+    :param val: val from which to gather end dates
+    :type val: :class:'~arelle.ModelXbrl.ModelXbrl'
+    :param role: The role for the relationship
+    :type role: str
+    :return: No direct return
+    :rtype: None
+    """
+    additional_axes = axis_config.get(_ADDITIONAL_AXES_KEY, {})
+    excluded_axes = axis_config.get(_EXCLUDED_AXES_KEY, {})
+    allowed_children = (
+        axis_config.get(_DEFINED_MEMBERS_KEY, []) +
+        axis_config.get(_ADDITIONAL_MEMBERS_KEY, [])
+    )
+    disallowed_children = []
+    disallowed_children.extend(
+        itertools.chain.from_iterable(
+            member_list for member_list in excluded_axes.values()
+        )
+    )
+    allowed_children.extend(
+        itertools.chain.from_iterable(
+            member_list for member_list in additional_axes.values()
+        )
+    )
+    if len(disallowed_children) > 0:
+        # Blacklisted axes check - Can only check blacklist (excluded)
+        # or whitelist (included) axes.
+        # Default to blacklist if both are present.
+        for child in _all_members_under(axis, relset):
+            if ((not _is_extension(child, val) and
+                 child.qname.localName in disallowed_children)):
+                fact = facts.axis_member_fact(
+                    axis.qname.localName,
+                    child.qname.localName,
+                    val.modelXbrl
+                )
+                if fact is not None:
+                    val.modelXbrl.error(
+                        '{base_key}.{extension_key}'.format(
+                            base_key=_CODE_NAME,
+                            extension_key=axis_config[_RULE_INDEX_KEY]
+                        ),
+                        messages.get_message(_CODE_NAME, _UGT_FACT_KEY),
+                        axis=axis.label(),
+                        member=child.label(),
+                        modelObject=fact,
```

Comment:
It takes different messages, as well as a group parameter.  I'd prefer to go with verbose but simple.